src/services/getTokenFromExchange.ts

```diff
@@ -0,0 +1,108 @@
+import { Erc20Api, TokenDetails, TokenList, ExchangeApi } from 'types'
+import { getImageUrl, log, getToken } from 'utils'
+import { getErc20Info } from './getErc20Info'
+
+interface TokenFromErc20Params {
+  tokenAddress: string
+  tokenId: number
+  erc20Api: Erc20Api
+}
+
+async function getTokenFromErc20(params: TokenFromErc20Params): Promise<TokenDetails | null> {
+  const { tokenAddress, tokenId, erc20Api } = params
+
+  // Get base info from the ERC20 contract
+  const erc20Info = await getErc20Info({ address: tokenAddress, erc20Api })
+  if (!erc20Info) {
+    log('Could not get details for token token (%s)', tokenAddress)
+    return null
+  }
+
+  const token = {
+    ...erc20Info,
+    id: tokenId,
+    address: tokenAddress,
+    image: getImageUrl(tokenAddress),
+  }
+
+  // TODO: add to tokenList?
```

Comment:
why? 

A getter should never add anything. Do you mean to the cache? If that's the case, we can handle this in other PR, but I would suggest adding the cache layer using the proxy pattern in the API layer (Cache implementation of the interface that performs a simple delegation).

Reply:
Yes, that's what I meant. Now that you mentioned I remembered about the proxy/cache layer we discussed. I'll update the TODOs to not forget again.

On a second note, @Velenir is what you said related to this?

Reply:
But remember we cache at API layer, not at service layer

Reply:
Yes

Reply:
https://github.com/gnosis/dex-react/issues/329